src/smart-components/role/add-role-new/add-permissions.js

```diff
@@ -0,0 +1,200 @@
+
+import React, { useState, useEffect } from 'react';
+import { shallowEqual, useSelector, useDispatch } from 'react-redux';
+import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components';
+import {
+    Table,
+    TableHeader,
+    TableBody,
+    TableVariant
+} from '@patternfly/react-table';
+import { getPrincipalAccess } from '../../../redux/actions/access-actions';
+import useFieldApi from '@data-driven-forms/react-form-renderer/dist/cjs/use-field-api';
+
+const columns = [ 'Application', 'Resource type', 'Operation' ];
+const selector = ({ accessReducer: { access, isLoading }}) => ({
+    access: access.data,
+    isLoading
+});
+const types = [ 'application', 'resource', 'operation' ];
+const accessWrapper = (rawData) => (filters = []) => {
```

Comment:
Phew this looks interesting, you should write some tests for it so we don't break anything crucial in future.